api/python/quilt3/packages.py

```diff
@@ -174,7 +174,7 @@ def set(self, path=None, meta=None):
             self
         """
         if path is not None:
-            self.physical_key = fix_url(path)
+            self.physical_key = PhysicalKey.from_url(fix_url(path))
```

Comment:
Should `from_url()` not automatically do `fix_url`?

Reply:
The problem with `fix_url` is, it's meant to handle ambiguous user input and guess what the user's intention is.

E.g., you can totally have a directory named `file:` or `s3:`, so _technically_, `s3://foo/bar` could mean, `./s3:/foo/bar`. But it's probably not what the user wants.
Also, _technically_, `c:` could be a URL scheme, but `c:/user/blah/` is probably just a path on Windows. And who knows what it should mean on Unix?

Anyways, I wanted to use `fix_url` only at the user-facing API boundaries, and never in the internal code.

Reply:
Okay, makes sense